### PR TITLE
Set 'reason' in EmailProgressed event and make 'friendlyDescription' …

### PR DESCRIPTION
**Pull request by fleetwoods**
…just that.

### CODE REVIEW

File: src/main/scala/com/ovoenergy/event/gateway/manager/api/resources/MailgunResource.scala [hunk 1]
```diff
@@ -137,8 +142,13 @@ class MailgunResource(clock: Clock,
...
                     formFields('custom, "Message-Id") { (customParameters, messageId) =>
                       complete {
-                        val description = form.getOrElse("description", "Failed event from mailgun")
-                        val rawData     = EmailProgressedData(description, customParameters, Some(messageId), Dropped)
+                        val rawData = EmailProgressedData(
+                          "Email failed status update received from Mailgun",
+                          customParameters,
+                          Some(messageId),
+                          Dropped,
+                          Some(form.getOrElse("description", "No failure reason supplied by Gateway"))
```
Comment (cb372) [inline src/main/scala/com/ovoenergy/event/gateway/manager/api/resources/MailgunResource.scala]: So "failed" events include the error description in the `description` field, while "bounced" events include it in the `error` field? Lovely.

**Comment by laurence-bird** [comment]
👍

File: src/main/scala/com/ovoenergy/event/gateway/manager/api/resources/MailgunResource.scala (hunk 1 above)
Comment (fleetwoods) [inline src/main/scala/com/ovoenergy/event/gateway/manager/api/resources/MailgunResource.scala]: Yeah and 'description' in failed events is not always present 👍

**Comment by ovo-comms-circleci** [comment]
This PR was successfully deployed to production by [CircleCI build 139](https://circleci.com/gh/ovotech/comms-gateway-event-manager/139) at 2017-04-28 11:43 BST.

**Pull request merged**